### ISSUE TITLE
Found runtime error on torch.masked_select(x, mask) has inconsistency bug between GPU and CPU testing

### ISSUE THREAD

**Issue by beanduan22** [pytorch/pytorch#113845]
### 🐛 Describe the bug

import torch
import torchvision
import torchvision.transforms as transforms
import torch.nn as nn
import os
import torch.nn.functional as F
gpu_device = torch.device("cuda")
cpu_device = torch.device("cpu")
class PreprocessAndCalculateModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.logsoftmax = nn.LogSoftmax(dim=1)
        self.flatten = nn.Flatten()
    def forward(self, x):
        # = x.view(x.size(0), -1
        mask = torch.rand_like(x) > 0.5
        x_masked = torch.masked_select(x, mask)
        return x_masked

transform = transforms.Compose([transforms.ToTensor()])
cifar_dataset = torchvision.datasets.CIFAR10(root='./data', train=True, transform=transform, download=True)
data_loader = torch.utils.data.DataLoader(cifar_dataset, batch_size=8, shuffle=True)

real_inputs, real_targets = next(iter(data_loader))
model = PreprocessAndCalculateModel()
x = real_inputs
output_gpu = model.to(gpu_device)(x.cuda())
output_cpu = model.to(cpu_device)(x.cpu())
print(torch.allclose(output_gpu.cpu(), output_cpu, atol=1e-1))

### Versions

2.0.0

**Comment by janeyx99** [comment]
Closing this issue as numerical accuracy differences are expected across devices, especially when mathematical operations compound. See https://pytorch.org/docs/stable/notes/numerical_accuracy.html. Usually, the difference is negligible and should not affect convergence. If that is not the case, please do not hesitate to follow up and reopen the issue!
